File: go/1010.go
package main

import "fmt"

func main() {
	var n1, n2 int
	var q1, q2, v1, v2 float64
	fmt.Scanf("%d", &n1)
	fmt.Scanf("%f", &q1)
	fmt.Scanf("%f", &v1)
	fmt.Scanf("%d", &n2)
	fmt.Scanf("%f", &q2)
	fmt.Scanf("%f", &v2)
	fmt.Printf("VALOR A PAGAR: R$ %.2f\n", q1*v1+q2*v2)
}

File: elixir/1010.ex
parse_array_as_floar = fn value ->
  value |> Enum.map(fn number -> Float.parse(number) |> elem(0) end)
end

calculate_total_row = fn item ->
  Enum.at(item, 1) * Enum.at(item, 2)
end

sum_rows = fn list_of_items ->
  list_of_items
  |> Enum.map(calculate_total_row)
  |> Enum.reduce(&(&1 + &2))
end

parse_mantisse = fn number, mantisse -> :erlang.float_to_binary(number, [decimals: mantisse]) end


item1 = IO.gets("") |> String.split
item2 = IO.gets("") |> String.split

items = [parse_array_as_floar.(item1) | [parse_array_as_floar.(item2)]]

result = parse_mantisse.(sum_rows.(items) ,2)

IO.puts("VALOR A PAGAR: R$ #{result}")
